### PR TITLE
Manual update dependencies from dotnet/arcade

### PR DESCRIPTION
**Pull request by f-alizada**
### Context
After merging the retargeting to dotnet 9 PR the build failed:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=10122195&view=results

Additionally automated updates: https://github.com/dotnet/msbuild/pull/10575
Also failing due to the version conflicts of Xunit.

### Changes Made
Commands: 
- `darc update-dependencies --channel ".NET Eng - Latest" --name "Microsoft.DotNet.Arcade.Sdk"`
- `darc update-dependencies --channel ".NET Eng - Latest" --name "Microsoft.SourceBuild.Intermediate.arcade"`
- `darc update-dependencies --channel ".NET Eng - Latest" --name "Microsoft.DotNet.XUnitExtensions"`

Temproary disabling the rules of xUnit. and will be solved as part of the: https://github.com/dotnet/msbuild/issues/10557

### Testing
- Existing tests should pass
- Build was tested on the exp insertion https://github.com/dotnet/msbuild/tree/exp/f-alizada/arcade-dependencies-retargeting

### CODE REVIEW

File: eng/common/core-templates/job/job.yml [hunk 1]
```diff
@@ -33,11 +33,6 @@ parameters:
   artifactPublishSteps: []
   runAsPublic: false
 
-# Sbom related params
```
[the remark this answers is not in the document]
Comment (f-alizada) [reply eng/common/core-templates/job/job.yml:36]: This is coming from common folder which is done in arcade: https://github.com/dotnet/arcade/pull/15014/files
Additionally we are passing the sbom generation parametr in our ci build: 
https://github.com/dotnet/msbuild/blob/a03ee4ba9c5e71a361d86a6ef7923c44a0dc5f39/.vsts-dotnet.yml#L147-L159

**Review by YuliiaKovalova — APPROVED** [review]

File: src/Build.UnitTests/BackEnd/BuildManager_Tests.cs [hunk 2]
```diff
@@ -1513,11 +1513,13 @@ public void CancelledBuildWithUnexecutedSubmission()
             _buildManager.EndBuild();
         }
 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```
Comment (rainersigwald) [inline src/Build.UnitTests/BackEnd/BuildManager_Tests.cs:1516]: Always nice to explain why:

```suggestion
#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously: needs to be async for xunit's timeout system
```

File: .editorconfig [hunk 3]
```diff
@@ -424,4 +424,11 @@ dotnet_diagnostic.IDE0100.severity = suggestion
...
-dotnet_diagnostic.IDE0330.severity = suggestion
\ No newline at end of file
+dotnet_diagnostic.IDE0330.severity = suggestion
+
+
+# xunit to supress temp
+dotnet_diagnostic.xUnit2020.severity = none
+dotnet_diagnostic.xUnit1031.severity = none
+dotnet_diagnostic.xUnit1012.severity = none
+dotnet_diagnostic.xUnit2029.severity = none
```
Comment (rainersigwald) [inline .editorconfig:434]: We have an existing `xUnit2013` suppression here, can you unify them in one place or the other?

https://github.com/dotnet/msbuild/blob/a03ee4ba9c5e71a361d86a6ef7923c44a0dc5f39/eng/Common.globalconfig#L1107-L1108

**Review by rainersigwald — APPROVED** [review]

File: .editorconfig (hunk 3 above)
Comment (f-alizada) [reply .editorconfig]: Moved to .editorconfig thank you for pointing to that

**Pull request merged**